modules/training.py

```diff
@@ -273,3 +302,25 @@ def threadedRun():
 def split_chunks(arr, step):
     for i in range(0, len(arr), step):
         yield arr[i:i + step]
+
+def cut_chunk_for_newline(chunk: str, max_length: int):
+    if '\n' not in chunk:
+        return chunk
+    first_newline = chunk.index('\n')
+    if first_newline < max_length:
+        chunk = chunk[first_newline + 1:]
+    if '\n' not in chunk:
+        return chunk
+    last_newline = chunk.rindex('\n')
+    if len(chunk) - last_newline < max_length:
+        chunk = chunk[:last_newline]
+    return chunk
+
+def format_time(seconds: float):
+    if seconds < 120:
+        return f"`{seconds:.0f}` seconds"
+    minutes = seconds / 60
+    if minutes < 120:
+        return f"`{minutes:.0f}` minutes"
+    hours = minutes / 60
+    return f"`{hours:.0f}` hours"
```

Comment:
With a little use of modulus here, we could display "h m s" instead of a single unit of time remaining
```suggestion
def format_time(seconds):
    # calculate hours, minutes, and seconds
    hours = seconds // 3600
    seconds %= 3600
    minutes = seconds // 60
    seconds %= 60
    
    # build the time string
    time_str = ""
    if hours:
        time_str += f"{hours}h "
    if hours || minutes:
        time_str += f"{minutes}m "
    time_str += f"{seconds}s"
    
    return time_str
```
Note: This is untested, just providing as a possible strategy.

Reply:
Looks valid, but the question is - which is simpler/clearer. I went with single number on the idea that `5h 23m 16s` isn't actually really providing much more info than `5 hours` is, but looks more complex.